src/js/component.js

```diff
@@ -1562,13 +1562,17 @@ class Component {
    *             return that if it exists.
    */
   static getComponent(name) {
-    if (!name) {
+    if (!name || !Component.components_) {
       return;
     }
 
+    if (Component.components_[name]) {
+      return Component.components_[name];
+    }
+
     name = toTitleCase(name);
```

Comment:
toTitleCase is one of the biggest offenders for startup time, since we `register` and `get` so many things. We should do anything that we can to prevent needing to call it.